device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarService.kt

```diff
@@ -270,12 +270,17 @@ public class CalendarService : PluginRegistry.RequestPermissionsResultListener {
             values.put(Events.EVENT_TIMEZONE, currentTimeZone.displayName);
 
             try {
-                val uri = contentResolver?.insert(CalendarContract.Events.CONTENT_URI, values);
+                var eventId: Long? = event.id?.toLongOrNull();
+                if (eventId == null) {
+                    val uri = contentResolver?.insert(CalendarContract.Events.CONTENT_URI, values);
+                    // get the event ID that is the last element in the Uri
+                    eventId = java.lang.Long.parseLong(uri?.getLastPathSegment());
+                } else {
+                    contentResolver?.update(ContentUris.withAppendedId(CalendarContract.Events.CONTENT_URI, eventId), values, null, null);
+                }
 
-                // get the event ID that is the last element in the Uri
-                val eventID = java.lang.Long.parseLong(uri?.getLastPathSegment());
 
-                finishWithSuccess(eventID.toString());
+                finishWithSuccess(eventId?.toString());
```

Comment:
should we be checking that eventId isn't null at this level instead of swallowing it or do you check at the Flutter/Dart layer?

Reply:
I thought I am handling this on the dart level. Turns out, I'm not. Will make sure that we are handling this scenario. Good catch